pkg/virt-controller/watch/drain/evacuation/evacuation.go

```diff
@@ -376,19 +376,29 @@ func (c *EvacuationController) sync(node *k8sv1.Node, vmisOnNode []*virtv1.Virtu
 
 	migrationCandidates, nonMigrateable := c.filterRunningNonMigratingVMIs(vmisToMigrate, activeMigrations)
 
+	activeMigrationsFromThisSourceNode := c.numOfVMIMForThisSourceNode(vmisOnNode, activeMigrations)
+
+	maxParallelMigrationsPerOutboundNode :=
+		int(*c.clusterConfig.GetMigrationConfiguration().ParallelOutboundMigrationsPerNode)
 	// Don't create hundreds of pending migration objects.
 	// This is just best-effort and is *not* intended to not overload the cluster.
 	// It is possible that more migrations than the limit are created because of evacuations on other nodes.
 	// The migration controller needs to limit itself to a reasonable number of running migrations
 	maxParallelMigrations := int(*c.clusterConfig.GetMigrationConfiguration().ParallelMigrationsPerCluster)
-	if len(activeMigrations) >= maxParallelMigrations {
-		// We have to re-enqueue if some work is left, since migrations from other controllers or workers` don't wake us up again
-		if len(migrationCandidates) > 0 || len(nonMigrateable) > 0 {
+
+	// No free slots, need to wait till migrations will finish. Re-enqueue and return
+	if len(migrationCandidates) > 0 || len(nonMigrateable) > 0 {
+		if activeMigrationsFromThisSourceNode >= maxParallelMigrationsPerOutboundNode ||
```

Comment:
nit: You could simplify this further and move this check after the free spots calculation. It would look like this:

```if freeSpots <= 0{ ...``` which subjectively reads better.

Reply:
@xpivarc Good catch! done.